core/client.go

```diff
@@ -121,118 +117,84 @@ func (c *Client) connect(ctx context.Context, addr string) error {
 		c.opts.Credential.Name(),
 		c.opts.Credential.Payload(),
 	)
-	err = c.WriteFrame(handshake)
-	if err != nil {
-		c.state = ConnStateRejected
+	if _, err := c.fs.WriteFrame(handshake); err != nil {
+		c.state = ConnStateDisconnected
 		return err
 	}
+
 	c.state = ConnStateConnected
 	c.localAddr = c.conn.LocalAddr().String()
 
 	c.logger.Printf("%s❤️  [%s][%s](%s) is connected to YoMo-Zipper %s", ClientLogPrefix, c.name, c.clientID, c.localAddr, addr)
 
 	// receiving frames
-	go c.handleFrame()
+	go func() {
+		closeConn, closeClient, err := c.handleFrame()
+		c.logger.Debugf("%shandleFrame: %v, %v, %T, %v", ClientLogPrefix, closeConn, closeClient, err, err)
+
+		c.mu.Lock()
+		defer c.mu.Unlock()
+
+		if c.state == ConnStateClosed {
+			return
+		}
+
+		c.state = ConnStateDisconnected
+		c.errc <- err
```

Comment:
In the BASIC example, if an error is returned here, the client will exit. In previous versions, the server dropped without an error。